src/payment/index.js

```diff
@@ -4,12 +4,22 @@ const grpc = require('@grpc/grpc-js')
 const protoLoader = require('@grpc/proto-loader')
 const health = require('grpc-js-health-check')
 const opentelemetry = require('@opentelemetry/api')
+const tracer = opentelemetry.trace.getTracer(process.env.OTEL_SERVICE_NAME);
```

Comment:
`process.env.OTEL_SERVICE_NAME` may be `undefined`, which is not a valid name for the tracer

Reply:
Ok thanks understood. I have switched to optional chain as suggested above

Reply:
So I guess I misundertood a little - Should we create the internal span? I think this makes more sense as we retain the original semantics

Reply:
All good, the way you did it is the way to go 🙂
Thanks for adapting it :)